tests/lib/rules/lines-around-comment.js

```diff
@@ -138,6 +301,38 @@ eslintTester.addRuleTest("lib/rules/lines-around-comment", {
             code: "bar()\n/**\n * block block block\n */\nvar a = 1;",
             options: [{ afterBlockComment: false, beforeBlockComment: true }],
             errors: [{ message: beforeMessage, type: "Block", line: 2 }]
+        },
+        {
+            code: "function foo(){\nvar a = 1;\n// line at block start\nvar g = 1;\n}",
+            options: [{
+                beforeLineComment: true,
+                allowBlockStart: true
+            }],
+            errors: [{ message: beforeMessage, type: "Line", line: 3 }]
+        },
+        {
+            code: "function foo(){\nvar a = 1;\n\n// line at block start\nvar g = 1;\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockEnd: true
+            }],
+            errors: [{ message: afterMessage, type: "Line", line: 4 }]
+        },
+        {
+            code: "while(true){\n// line at block start and end\n}",
+            options: [{
+                afterLineComment: true,
+                allowBlockStart: true
+            }],
+            errors: [{ message: afterMessage, type: "Line", line: 2 }]
+        },
+        {
+            code: "while(true){\n// line at block start and end\n}",
```

Comment:
I'm a bit confused now, when dealing with comment-only blocks and when only one of either `beforeLineComment` and `afterLineComment` are set, are only one of `allowBlockStart` and `allowBlockEnd` (respectively) necessary for the lint to pass?

If so, of the test cases I showed above, only the third one would be relevant. (And this one would need to be added:  https://github.com/eslint/eslint/pull/2741/files#r32274807)